app/api/api_v1/endpoints/search.py

```diff
@@ -1,39 +1,81 @@
+
+
 @router.get("/", response_model = schemas.SearchResultFull)
 def search(query: str) -> Any:   
-    emptySearchResult =  schemas.SearchResult(id="", title="", description="", path="", locale="")
-    results = []
-    suggestions = []        # EMPTY FOR NOW
+    # A dictionary to store top_k results for each question across all collections
+    top_results = {}
 
+    question_results = []
+    results = []
+    answers = []
+    links = []
+    # Generate questions from the query
+    questions = utils.generate_questions(query, n = utils.N_QUESTIONS)
+    existing_collections = [x.decode('utf-8') for x in deps.db.connection.execute_command('FT._LIST')]
+    #print("Find Existing Collections!")
     if len(existing_collections) < 1:
         raise HTTPException(status_code=404, detail="No Collections")
 
-    for collection_name, collection in existing_collections.items():
-        if len(collection.documents) < 1:
-            continue
-        for document_id, document in collection.documents.items():
-            if query.lower() == document.title.lower():
-                result = schemas.SearchResult(
-                    id=f"{collection_name}_{document_id}",  # ID IS PATH FOR NOW
-                    title=document.title,
-                    description=document.description,
-                    path=document_id,
-                    locale=document.localeCode
-                )
-                results.append(result)
-
-    return schemas.SearchResultFull(
-        results=results,
-        suggestions=suggestions,
-        totalHits=len(results) + len(suggestions)
-    )
+    for question in questions:
+        query_embedding = utils.encode_blocks(deps.db.embedding, question)
+        # A list to store the search results for the current question from all collections
+        for collection in existing_collections:
+            #print(f"Search in {collection} ")
+            redis_search_results = index.search_vectors_knn(redis_connect = deps.db.connection, 
+                                                            query_vector = query_embedding, 
+                                                            index_name = collection, 
+                                                            vector_name = utils.VECTOR_FIELD_NAME, 
+                                                            top_k = utils.TOP_LINKS)
+            #print(f"print 1 hit: {redis_search_results}")
+            if redis_search_results is not None:
+                question_results.extend(redis_search_results)
+                #print(f"Add result from {collection}")
+
+        # Sort the search results for the current question based on the scores and select top_k results
+        sorted_list = sorted(question_results, key=lambda x: float(x['dist']))
+        top_results[question] = sorted_list[:utils.TOP_LINKS]
+        #print(' Store best 5 for a question!')
 
-    
\ No newline at end of file
+        for redis_doc in top_results[question]:
```

Comment:
Please explore `redis_om` and consider to replace that unreadable `redis_path = redis_dict['path'.encode('utf-8')].decode('utf-8') …` into `block = DocumentBlock.get( result.id)`  and you’ll get access to `block.path` using [redis_om](https://github.com/redis/redis-om-python).